### DISCUSSION TITLE
TKB TZ06: Switch state not automatically received by ZwaveJS - manual refresh required

### DISCUSSION THREAD

**Discussion by BillsGate** [zwave-js/zwave-js#3151] [Q&A] [answered]
I'm using ZwaveJS in HomeAssistant and among all my Z-wave devices I own a couple of Zipato PH-pan04 double switch modules, which are in fact rebrands of TKB TZ06 modules. They show up as TKB TZ06 in my ZwaveJS-panel.

Since a few updates of ZwaveJS, the switch state of these modules isn't received by ZwaveJS: If I push a physical switch connected to the module, the light turns on, but this state-change isn't updated in the _Binary switch_-panel of the module in ZwaveJS

![image](https://user-images.githubusercontent.com/1685888/128154472-fd9b0cbb-0cc3-4427-9289-9b2094375415.png)

Pressing the _REFRESH_ button at the right side of the panel updates the values correctly, but having to refresh manually defeats the purpose of automating switch actions :)

Other Zwave-devices in my network don't have this issue.

Is there any way to make the device-refresh in ZwaveJS happen automatically for these modules? 
Am I doing something wrong?
Is there some logging which I can use to debug the failing refresh?

**Comment by AlCalzone** [comment, marked as answer]
Can you open this as a bug report please and attach a driver log where you re-interview the device and then toggle the switch afterwards (all channels please if there are multiple)?

In v8 we removed the automatic mapping of reports from the root device to endpoint 1, which might be the reason you're seeing this.

**Reply by BillsGate** [reply]
Thank you, I've submitted a [bug report](https://github.com/zwave-js/node-zwave-js/issues/3153)

Pressing physical switches doesn't log anything, even in loglevel Silly
